src/engine/sparqlExpressions/StringExpressions.cpp

```diff
@@ -399,27 +399,27 @@ class ConcatExpression : public detail::VariadicExpression {
 using EncodeForUriExpression =
     StringExpressionImpl<1, decltype(encodeForUriImpl)>;
 
-// TO_NUMERIC (int or double)
-auto InvalidArg = std::errc::invalid_argument;
-auto OutOfRange = std::errc::result_out_of_range;
-
+// Expressions that convert a string to a number (int or double).
 template <typename T>
 [[maybe_used]] auto toNumeric = [] (std::optional<std::string> input) {
   if (!input.has_value()) {
     return Id::makeUndefined();
   } else {
-    T res{};
-    auto str = boost::lexical_cast<std::string>(*input);
-    auto conv = std::from_chars(str.data(), str.data() + str.size(), res);
-    if (conv.ec == InvalidArg || conv.ec == OutOfRange) {
+    auto str = absl::StripAsciiWhitespace(input.value());
+    double resD{};
```

Comment:
You changed the behavior for `xsd:integer("42.31")` (Edit: you didn't)
I didn't find the appropriate standardization, but other engines make this an UNDEFINED value, which I can agree with.
So maybe we should leave this a `T res;` for now without the `static_cast`.

Reply:
T res will throw an error because absl::from_chars() doesn't allow directly binding the value to integer. https://github.com/abseil/abseil-cpp/blob/master/absl/strings/charconv.h (lines 52 -53).

Reply:
Ah, `absl::from_chars` ALWAYS assumes a double....
You then have to use `std::from_chars` for integer values like so:

```
T res;
if constexpr ( isIntegral) { std::from_chars(...)} else {absl::from_chars}.
```
Unfortunately this gives us very much duplication, but this is probably the cleanest way.